backend/models/core/EntityType.js

```diff
@@ -56,75 +65,94 @@ class EntityTypes {
-      ]);
 
-      //TODO: return something valuable here
-      return true;
-
-    }
-
-
-    static async create({name, slug}) {
-
-      const db = new DB();
-
-      const insertEntitiesSQL = 'INSERT into entity_types (slug, name) VALUES (:slug, :name)';
-      
-      await db.executeStatement(insertEntitiesSQL, [
-        {name: 'slug', value:{stringValue: slug}},
-        {name: 'name', value:{stringValue: name}},
-      ]);
-
-      //TODO: return something valuable here
-      return true;
-
-    }
-
-    static async delete({slug}) {
-      const db = new DB();
-      const deleteEntityTypesSQL = 'DELETE from entity_types where slug = :slug'
-  
-      let executeStatementParam = [
-        {name: 'slug', value:{stringValue: slug}}
-      ]
-      await db.executeStatement(deleteEntityTypesSQL, executeStatementParam);
-  
-      //TODO: return something valuable here
-      return true;
-    }
+    const data = await db.executeStatement(sql, [
+      { name: "slug", value: { stringValue: slug } },
+    ]);
+
+    return data.records.map(
+      ([
+        { longValue: entity_type_id },
+        { stringValue: entity_type_name },
+        { stringValue: entity_type_slug },
+        { stringValue: attribute_name },
+        { stringValue: attribute_type },
+        { longValue: attribute_order },
+        { longValue: attribute_id },
+      ]) => ({
+        entity_type_id,
+        entity_type_name,
+        entity_type_slug,
+        attribute_name,
+        attribute_type,
+        attribute_order,
+        attribute_id,
+      })
+    );
+  }
+
+  static async whereSlug({ slug }) {
+    return this.find({ slug });
+  }
+
+  static async create({ name, slug }) {
+    const db = new DB();
+
+    const insertEntitiesSQL =
+      "INSERT into entity_types (slug, name) VALUES (:slug, :name)";
+
+    await db.executeStatement(insertEntitiesSQL, [
+      { name: "slug", value: { stringValue: slug } },
+      { name: "name", value: { stringValue: name } },
+    ]);
+
+    //TODO: return something valuable here
+    return true;
+  }
+
+  static async create({ name, slug }) {
+    const db = new DB();
+
+    const insertEntitiesSQL =
+      "INSERT into entity_types (slug, name) VALUES (:slug, :name)";
+
+    await db.executeStatement(insertEntitiesSQL, [
+      { name: "slug", value: { stringValue: slug } },
+      { name: "name", value: { stringValue: name } },
+    ]);
+
+    //TODO: return something valuable here
+    return true;
+  }
+
+  static async delete({ slug }) {
+    const db = new DB();
+    const deleteEntityTypesSQL = "DELETE from entity_types where slug = :slug";
+
+    let executeStatementParam = [
+      { name: "slug", value: { stringValue: slug } },
+    ];
+    await db.executeStatement(deleteEntityTypesSQL, executeStatementParam);
+
+    //TODO: return something valuable here
+    return true;
+  }
+
+  static async update({ name, newSlug, oldSlug }) {
+    const db = new DB();
+    const updateEntityTypesSQL =
```

Comment:
Please remove the new line on line 143. 
```suggestion
    const updateEntityTypesSQL =  "UPDATE entity_types SET slug = :newSlug, name = :name WHERE slug = :oldSlug";
```